base/strings/io.jl

```diff
@@ -427,7 +427,7 @@ function unescape_string(io::IO, s::AbstractString, keep = ())
                 m = c == 'x' ? 2 :
                     c == 'u' ? 4 : 8
                 while (k += 1) <= m && !isempty(a)
-                    nc = peek(a)
+                    nc = peek(a)::Char
```

Comment:
Can you be sure this is a `Char` (thinking about `s` being an `AbstractString`). Maybe needs `AbstractChar` or `Char(peek(a))::Char`?

Reply:
thanks ! yeah we should change this to `AbstractChar`